core/src/main/java/org/elasticsearch/search/aggregations/metrics/avg/ParsedAvg.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.metrics.avg;
+
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.metrics.ParsedNumericMetricsAggregation;
+
+import java.io.IOException;
+
+public class ParsedAvg extends ParsedNumericMetricsAggregation.SingleValue implements Avg {
+
+    @Override
+    public double getValue() {
+        return value();
+    }
+
+    @Override
+    protected String getType() {
+        return AvgAggregationBuilder.NAME;
+    }
+
+    private static final ObjectParser<ParsedAvg, Void> PARSER = new ObjectParser<>(AvgAggregationBuilder.NAME, true, ParsedAvg::new);
```

Comment:
I don't have a strong opinion here, I don't think it makes a whole lot of difference